src/client/dfs/dfs.c

```diff
@@ -1123,6 +1138,11 @@ dfs_cont_create(daos_handle_t poh, uuid_t co_uuid, dfs_attr_t *attr,
 	else
 		dattr.da_oclass_id = DFS_DEFAULT_OBJ_CLASS;
 
+	if (attr && attr->da_mode == DFS_RELAXED)
+		dattr.da_mode = attr->da_mode;
+	else
+		dattr.da_mode = DFS_BALANCED;
+
```

Comment:
Do we need to use `MODE_MASK` here just in case some other bits were set?

Reply:
> maybe we need to fail in case it's set to some rubbish

I think that might be a good idea. I'm mostly considering what might happen if we _**do**_ happen to add some other flag  in the future and this line is overlooked.
Or if somehow we just accidentally reference/set the wrong variable at some point, and then this line overlooks that.

Reply:
ok updated to check for that bit, and use balanced in default case.
the latter would still need some changes if we add another mode/flag anyway, but maybe this is clearer.

Reply:
Yes, I think the update is much clearer. Thanks!